libsolidity/formal/BMC.cpp

```diff
@@ -478,6 +478,11 @@ void BMC::internalOrExternalFunctionCall(FunctionCall const& _funCall)
 	auto const& funType = dynamic_cast<FunctionType const&>(*_funCall.expression().annotation().type);
 	if (shouldInlineFunctionCall(_funCall))
 		inlineFunctionCall(_funCall);
+	else if (isPublicGetter(&_funCall.expression()))
+	{
+		// DO NOTHING HERE, the processing happens in SMT Encoder, but we need to prevent resetting the state variables
+//		visitPublicGetter(_funCall);
```

Comment:
Can remove this line

Reply:
I believe this is related to my hesitance of where to put the call to `visitPublicGetter`.
This seemed like a more natural place for BMC then the one where it called now. But we would have to find a similar good place for CHC.